src/main/java/sorald/api/RuleRepository.java

```diff
@@ -0,0 +1,48 @@
+package sorald.api;
+
+import java.util.Collection;
+import javax.annotation.Nonnull;
+import sorald.rule.IRuleType;
+import sorald.rule.Rule;
+
+/**
+ * This class defines the interface for a repository of rules. It is used by Sorald to find all
+ * rules to use. All implementing classes need a no-arg constructor.
+ */
+public interface RuleRepository {
```

Comment:
Is there some difference between `RuleRepository` and `Rules`? I think we can get rid of the latter as `RuleRepository` defined the behaviour and `SonarRules` implements it.

Reply:
Rules is the aggregation class providing an API to query all repositories. We have multiple `RuleRepository` in the future.

Reply:
Sounds good. Can we keep a better name so that it is clearer? I can't think of any.

Reply:
`RuleProvider` would fit the naming scheme imo. Then we have multiple repository containing data and a single provider to query against.